src/cpp/CMakeLists.txt

```diff
@@ -130,14 +130,17 @@ if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
    set(Boost_USE_STATIC_LIBS ON)
 endif()
 
-# default to boost 1.50.0
-set(BOOST_VERSION 1.50.0)
+# default to boost 1.63.0
+set(BOOST_VERSION 1.63.0)
 
 # override if necessary
 if (RSTUDIO_BOOST_VERSION)
    set(BOOST_VERSION "${RSTUDIO_BOOST_VERSION}")
 endif()
 
+# disable Boost Signals deprecation warning
+add_definitions(-DBOOST_SIGNALS_NO_DEPRECATION_WARNING)
```

Comment:
We might want to consider switching to `Boost.Signal2` as the original `Boost.Signals` library is now deprecated. (IIUC this was done because `Boost.Signals` is not thread-safe, but we never actually use Boost signals for cross-thread communication so this in theory doesn't affect us)